api-admin/src/main/java/bio/terra/pearl/api/admin/service/study/StudyEnvironmentExtService.java

```diff
@@ -51,40 +55,86 @@ public StudyEnvironment update(PortalStudyEnvAuthContext authContext, StudyEnvir
-    return new StudyEnvStats(
-        enrolleeService.countByStudyEnvironmentId(studyEnv.getId()),
-        withdrawnEnrolleeService.countByStudyEnvironmentId(studyEnv.getId()));
+  @EnforcePortalStudyEnvPermission(permission = "study_settings_edit")
+  public void updateKitTypes(PortalStudyEnvAuthContext authContext, List<String> updatedKitTypes) {
+    List<KitType> allowedKitTypes = studyEnvironmentKitTypeService.findAllowedKitTypes();
+    List<KitType> configuredKitTypes =
+        studyEnvironmentKitTypeService.findKitTypesByStudyEnvironmentId(
+            authContext.getStudyEnvironment().getId());
+
+    validateNoRemovalOfKitTypes(updatedKitTypes, configuredKitTypes);
+    validateKitTypes(updatedKitTypes, allowedKitTypes);
+
+    List<StudyEnvironmentKitType> newKitTypes = new ArrayList<>();
+    for (String kitTypeName : updatedKitTypes) {
+      KitType kitType =
+          allowedKitTypes.stream()
+              .filter(k -> k.getName().equals(kitTypeName))
+              .findFirst()
+              .orElseThrow(() -> new BadRequestException("Invalid kit type"));
+      StudyEnvironmentKitType studyEnvironmentKitType =
+          StudyEnvironmentKitType.builder()
+              .studyEnvironmentId(authContext.getStudyEnvironment().getId())
+              .kitTypeId(kitType.getId())
+              .build();
+      newKitTypes.add(studyEnvironmentKitType);
+    }
+
+    studyEnvironmentKitTypeService.bulkCreate(newKitTypes);
   }
 
-  private StudyEnvironment authToStudyEnv(
-      AdminUser operator, String portalShortcode, String studyShortcode, EnvironmentName envName) {
-    authUtilService.authUserToStudy(operator, portalShortcode, studyShortcode);
-    return studyEnvService.findByStudy(studyShortcode, envName).get();
+  @EnforcePortalStudyEnvPermission(permission = AuthUtilService.BASE_PERMISSON)
+  public List<KitType> getAllowedKitTypes(PortalStudyEnvAuthContext authContext) {
+    return studyEnvironmentKitTypeService.findAllowedKitTypes();
+  }
+
+  @EnforcePortalStudyEnvPermission(permission = AuthUtilService.BASE_PERMISSON)
+  public StudyEnvStats getStats(PortalStudyEnvAuthContext authContext) {
+    return new StudyEnvStats(
+        enrolleeService.countByStudyEnvironmentId(authContext.getStudyEnvironment().getId()),
+        withdrawnEnrolleeService.countByStudyEnvironmentId(
+            authContext.getStudyEnvironment().getId()));
   }
 
   public record StudyEnvStats(int enrolleeCount, int withdrawnCount) {}
+
+  protected void validateNoRemovalOfKitTypes(
+      List<String> updatedKitTypes, List<KitType> existingKitTypes) {
+    Set<String> updatedKitTypeSet = new HashSet<>(updatedKitTypes);
+    Set<String> existingKitTypeSet =
+        existingKitTypes.stream().map(KitType::getName).collect(Collectors.toSet());
+
+    if (!updatedKitTypeSet.containsAll(existingKitTypeSet)) {
+      throw new BadRequestException("You may not remove a kit type from a study environment");
+    }
+  }
```

Comment:
Should we check to see if any kits of this type exist and only under those circumstances not allow deletion? I could see someone misclicking then not being able to remove it.

Also, I think this error should be an `IllegalArgumentException` - `BadRequest` triggers a 500 so the user doesn't see the error, so I was a bit confused at first. There's a few other usages of BadRequestException in this file that I'm not sure should trigger 500s also

Reply:
concur. with IllegalArgument instead of BadRequest

Reply:
you could either check that no kits of that type exist, or check that that no types that are configured for prod have been removed.  that way someone can experiment with types in sandbox

Reply:
I thought about this more and I don't think there's actually a good reason to restrict kit type removal at all now that it's going through the publishing infrastructure.

There aren't any dangling FK constraints to worry about either as the actual kit requests reference the `kit_type` table, not the `study_environment_kit_type` table (which I think is right). Since the `kit_type`s are global and populated by the seed kitTypes.json file, those shouldn't be going anywhere